src/lib/security/selinux_config.rb

```diff
@@ -25,92 +25,156 @@ module Security
   class SelinuxConfig
     include Yast::Logger
     Yast.import "Bootloader"
+    Yast.import "ProductFeatures"
 
     attr_reader :policy
 
+    # Constructor
     def initialize
-      propose_policy if Yast::Mode.installation
-      @policy = configured_policy
+      @policy = initial_policy
+    end
+
+    # Return the initial SELinux policy
+    #
+    # It can be the proposed one if running in an installation or the configured policy for a
+    # running system
+    #
+    # @return [Symbol]
+    def initial_policy
+      @initial_policy ||=
+        begin
+          propose_policy if Yast::Mode.installation
+          configured_policy
+        end
     end
 
     # Returns the policy applied in the running system
+    #
+    # @note the system can be booted with a different SELinux policy that the configured one. To
+    #  know the running policy getenforce tool is used.
+    #
+    # @return [Symbol] running SELinux policy if command executed successfully; :disabled otherwise
     def running_policy
       Yast::Execute.locally!(GETENFORCE_PATH, stdout: :capture).chomp.downcase.to_sym
     rescue Cheetah::ExecutionFailed
       log.debug("`#{GETENFORCE_PATH}` not available. SELinux or selinux-tools missing")
       :disabled
     end
 
-    # If current policy value is different to the configured one
-    def changed?
-      @policy != configured_policy
+    # Set the policy to given value
+    #
+    # @note using nil means to set SELinux policy as disabled.
+    #
+    # @param id [String, Symbol, nil] a SELinux policy identifier
+    # @return [] true when policy found and set; false otherwise
+    def policy=(id)
+      @policy = id&.to_sym || :disabled
     end
 
-    # Use the given policy for the next boot (in running system is needed to #save)
-    def update_policy(policy_key)
-      found_policy = find_policy(policy_key)
-
-      if found_policy
-        log.info("Changing SELinux configuration to `#{policy_key}` policy: #{found_policy}")
-
-        @policy = policy_key
-      else
-        log.info("Unknown `#{policy_key}` SELinux policy")
-      end
-    end
-    alias_method :policy=, :update_policy
-
+    # Set current policy options as kernel parameters for the next boot
+    #
+    # @note it does not write the changes when running in installation mode, where only sets the
+    #   kernel params in memory, since the Yast::Bootloader.Write will be performed at the end of
+    #   installation.
+    #
+    # @see Yast::Bootloader#modify_kernel_params
+    #
+    # @return [Boolean] false if there area not policy options or nothing changed; true otherwise
```

Comment:
typo area. And I also do not like much case that wrong policy and not changed policy has same return state.